### PR TITLE
added unit tests for transfer request exit service

### PR DESCRIPTION
**Pull request by lodug**
Added logger
Added modifications in the Makefile
Unit tests

### CODE REVIEW

File: src/ota/request_transfer_exit/utest/RequestTransferExitTest.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+{
+    /* simulate a successfull transfer data*/
+    bool transferSuccess =  true;
+    /** Set a lambda function as the callback for the RequestTransferExit class that takes 
+    *   one parameter of type boolean named and the body of lambda simply returns the success parameter
+    */
+    transferExit->setTransferCompleteCallBack([](bool success) { return success; });
+
+    bool result = transferExit->requestTransferExit(0x10, transferSuccess);
+    /*  */
```
Comment (RubenNatanael) [inline src/ota/request_transfer_exit/utest/RequestTransferExitTest.cpp:44]: Remove comment, also in line 55

File: src/ota/request_transfer_exit/utest/RequestTransferExitTest.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
+#include <gtest/gtest.h>
+#include "../include/RequestTransferExit.h"
+
+/* Class fixture for general tests */
+class RequestTransferExitTest : public ::testing::Test
+{
+protected:
+    Logger logger;
+    RequestTransferExit* transferExit;
```
Comment (RubenNatanael) [inline src/ota/request_transfer_exit/utest/RequestTransferExitTest.cpp:9]: Name variables with all lowercase and underscores between words

File: src/ota/request_transfer_exit/utest/RequestTransferExitTest.cpp (hunk 1 above)
Comment (lodug) [reply src/ota/request_transfer_exit/utest/RequestTransferExitTest.cpp]: done

File: src/ota/request_transfer_exit/utest/RequestTransferExitTest.cpp (hunk 2 above)
Comment (lodug) [reply src/ota/request_transfer_exit/utest/RequestTransferExitTest.cpp]: done

File: src/ecu_simulation/BatteryModule/Makefile [hunk 3]
```diff
@@ -1,11 +1,11 @@
 #Compiler flags
-CFLAGS = -std=c++17 -g -Wall -pthread -I./include -I./src -I../../utils/include -I../../utils/src -I../../uds/include -I../../uds/src
```
Comment (Mihnea9901) [inline src/ecu_simulation/BatteryModule/Makefile:2]: restore -std-c++17 flag

File: src/ecu_simulation/BatteryModule/Makefile [hunk 4]
```diff
@@ -75,13 +75,18 @@ GenerateFrames_test: ../../utils/test/GenerateFramesTest.cpp ../../utils/src/Gen
 # CreateInterface Unit tests
 createInterfaceTest: CreateInterface_test
 CreateInterface_test: ../../utils/test/CreateInterface_test.cpp ../../utils/src/CreateInterface.cpp
-	$(CXX) $(CFLAGS) $(CFLAGSTST) -o ../../utils/test/CreateInterface_test ../../utils/test/CreateInterface_test.cpp ../../utils/src/CreateInterface.cpp ../../utils/src/Logger.cpp $(CFLAGSTST2) $(LDFLAGS)
+	$(CXX) $(CFLAGS) $(CFLAGSTST) -o ../../utils/test/CreateInterface_test ../../utils/test/CreateInterface_test.cpp $(SRCSTEST) $(CFLAGSTST2) $(LDFLAGS)
```
Comment (Mihnea9901) [inline src/ecu_simulation/BatteryModule/Makefile:78]: restore this test

File: src/ecu_simulation/BatteryModule/Makefile [hunk 5]
```diff
@@ -158,7 +163,7 @@ GenerateFramesCoverage: ../../utils/test/GenerateFramesTest.cpp ../../utils/src/
 # CreateInterface Coverage
 createInterfaceCoverage: CreateInterfaceCoverage
 CreateInterfaceCoverage: ../../utils/test/CreateInterface_test.cpp ../../utils/src/CreateInterface.cpp
-	  $(CXX) $(CFLAGS) $(CFLAGSTST) -fprofile-arcs -ftest-coverage -o ../../utils/test/CreateInterface_test ../../utils/test/CreateInterface_test.cpp ../../utils/src/CreateInterface.cpp ../../utils/src/Logger.cpp $(CFLAGSTST2) $(LDFLAGS)
+	  $(CXX) $(CFLAGS) $(CFLAGSTST) -fprofile-arcs -ftest-coverage -o ../../utils/test/CreateInterface_test ../../utils/test/CreateInterface_test.cpp $(SRCSTEST) $(CFLAGSTST2) $(LDFLAGS)
```
Comment (Mihnea9901) [inline src/ecu_simulation/BatteryModule/Makefile:166]: also restore this

File: src/ecu_simulation/BatteryModule/Makefile [hunk 6]
```diff
@@ -220,7 +242,7 @@ GenerateFrames_coverage:
 	mv GenerateFrames.cpp.gcov coverage/out_GenerateFrames
 
 CreateInterface_coverage:
-	$(CXX) $(CFLAGS) $(CFLAGSTST) -fprofile-arcs -ftest-coverage -o ../utils/test/CreateInterface_test ../utils/test/CreateInterface_test.cpp ../../utils/src/CreateInterface.cpp ../../utils/src/Logger.cpp $(CFLAGSTST2) $(LDFLAGS)
+	$(CXX) $(CFLAGS) $(CFLAGSTST) -fprofile-arcs -ftest-coverage -o ../utils/test/CreateInterface_test ../utils/test/CreateInterface_test.cpp $(SRCSTEST) $(CFLAGSTST2) $(LDFLAGS)
```
Comment (Mihnea9901) [inline src/ecu_simulation/BatteryModule/Makefile:245]: also restore this

File: src/ecu_simulation/BatteryModule/Makefile (hunk 3 above)
Comment (lodug) [reply src/ecu_simulation/BatteryModule/Makefile]: done

File: src/ecu_simulation/BatteryModule/Makefile (hunk 4 above)
Comment (lodug) [reply src/ecu_simulation/BatteryModule/Makefile]: done

File: src/ecu_simulation/BatteryModule/Makefile (hunk 5 above)
Comment (lodug) [reply src/ecu_simulation/BatteryModule/Makefile]: done

File: src/ecu_simulation/BatteryModule/Makefile (hunk 6 above)
Comment (lodug) [reply src/ecu_simulation/BatteryModule/Makefile]: done

**Review by RubenNatanael — APPROVED** [review]

**Review by Mihnea9901 — APPROVED** [review]

**Pull request merged**